sparkctl/cmd/create.go

```diff
@@ -267,28 +272,118 @@ func isContainerLocalFile(file string) (bool, error) {
 	return false, nil
 }
 
+type Blob interface {
+	// TODO: With go-cloud supporting setting ACLs, remove implementations of interface
+	setPublicACL(ctx context.Context, bucket string, filePath string) error
+}
+
+type UploadHandler struct {
+	blob             Blob
+	blobUploadBucket string
+	blobEndpoint     string
+	hdpScheme        string
+	ctx              context.Context
+	b                *blob.Bucket
+}
+
+func (uh UploadHandler) uploadToBucket(uploadPath, localFilePath string) (string, error) {
+	fileName := filepath.Base(localFilePath)
+	uploadFilePath := filepath.Join(uploadPath, fileName)
+	// Check if exists by trying to fetch metadata
+	_, err := uh.b.NewRangeReader(uh.ctx, uploadFilePath, 0, 0)
```

Comment:
Sorry if I didn't make it clear. You need to close the returned `Reader`, but you don't need to use `defer reader.Close()`. Simply do the following is fine.
```
reader, err := uh.b.NewRangeReader(uh.ctx, uploadFilePath, 0, 0)
reader.Close()
```

Reply:
Oh, ok - now I got your point.